docs/rfcs/0001-designing-in-the-open.md

```diff
@@ -0,0 +1,142 @@
+* Lower the barrier to entry for interested parties to become maintainers
+
+## Non-Goals
+
+* Enforce that every change originates from a fully public proposal or
+  discussion. While all maintainers and contributors will be encouraged to
+  design openly, there may be legal, security, privacy, or business reasons
+  that prevent some or all context from being made public.
+
+* Be overly prescriptive: too many rules can hinder adoption of a process. This
+  document outlines intention, not hard policy.
+
+## Proposal
+
+Public proposals may take one of two forms:
+
+* Issue proposals
+* RFC PR proposals (e.g., this document)
+
+### Issues
+
+Issues are the quickest path towards proposing a change. Issue proposals must
+be opened at the [grafana/agent issues page](https://github.com/grafana/agent/issues).
+
+There are no strict set of rules for issue-based proposals, but authors are
+recommended to prefix the issue title with `Proposal:` so it may be found more
+easily.
+
+### RFC PRs
+
+RFC PR proposals must at least:
+
+* Be placed in the `docs/rfcs` folder of the `grafana/agent` repository
+* Have a lowercase filename in hyphen-case with an `.md` extension
+* Prefix the filename with the RFC ID
+  * ID `xxxx` may be initially used until the final ID is known
+* Contain valid markdown
+* Start with the title of the proposal
+* Contain a bullet point list of metadata of:
+  * The date the proposal was written
+  * The list of authors, with their names and GitHub usernames
+  * The PR where the proposal was posted
+
+`0000-template.md` contains a template to use for writing proposals that
+conforms to these rules.
+
+The remainder of the proposal may be formatted however the author wishes. Some
+example sections in the RFC may be:
+
+* Summary: What is the background that lead to this proposal?
+* Goals: What are the main goals of the proposal?
+* Non-Goals: What _aren't_ the main goals of the proposal?
+* Proposal: What is the proposal?
+* Pros/Cons:
+  * Pros: What are the upsides to this proposal?
+  * Cons: What are the downsides to this proposal?
+* Considered Alternatives: Why is this proposal the best path forward? What
+  were the alternatives?
+* Open Questions: What questions still need to be answered?
+* Prior Art: What was this proposal based on, if anything?
+
+#### RFC Review
+
+RFCs should be opened as a PR to grafana/agent, ideally prefixed in the PR
+title with `RFC:` to easily identify it amongst other PRs.
+
+### Google Docs Proposals
+
+Google Docs may be useful for early feedback rounds during a proposal. However,
+it is not recommended for the permanent home of a proposal:
+
+* Change and comment history may not be available to all viewers.
+
+* The file owner may delete the proposal, leading to a gap in historical
+  context.
+
+Google Docs proposals will be permitted if linked to from an issue proposal.
+
+It is recommended that proposals from Google Docs are eventually converted by
```

Comment:
I think this should be required. The repo should be the source of truth.

Reply:
Good idea, let me know what you think of the rephrased wording.